### DISCUSSION TITLE
How does AGE optimize queries?

### DISCUSSION THREAD

**Discussion by TheOtherBrian1** [apache/age#1784] [Q&A] [answered]
Does it create specialized indexes or is it primarily syntactic sugar for writing complex PostgreSQL recursive queries in Cypher?

**Comment by markgomer** [comment]
@jrgemignani @rafsun42 @dehowef @MuhammadTahaNaveed, could you give a hand here?

**Comment by jrgemignani** [comment, marked as answer]
@TheOtherBrian1 I wish it was all syntactic sugar, that would have made this project a lot easier :) Cypher queries are transformed into queries that PG can understand or process with its own backend. We try to use as much of PGs framework as possible so as to enable usage of things like indexes, for example.
